src/coreclr/vm/precode_portable.cpp

```diff
@@ -0,0 +1,210 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+
+#include "common.h"
+#include "precode_portable.hpp"
+
+#ifdef HOST_64BIT
+    #define CANARY_VALUE 0x1234567812345678
+#else // HOST_64BIT
+    #define CANARY_VALUE 0x12345678
+#endif // HOST_64BIT
+
+bool PortableEntryPoint::IsNativeEntryPoint(TADDR addr)
+{
+    STANDARD_VM_CONTRACT;
+
+    return false;
+}
+
+MethodDesc* PortableEntryPoint::GetMethodDesc(TADDR addr)
+{
+    STANDARD_VM_CONTRACT;
+
+    PortableEntryPoint* portableEntryPoint = ToPortableEntryPoint(addr);
+    _ASSERTE(portableEntryPoint->_pMD != NULL);
+    return portableEntryPoint->_pMD;
+}
+
+void* PortableEntryPoint::GetInterpreterData(TADDR addr)
+{
+    STANDARD_VM_CONTRACT;
+
+    PortableEntryPoint* portableEntryPoint = ToPortableEntryPoint(addr);
+    _ASSERTE(portableEntryPoint->_pInterpreterData != NULL);
+    return portableEntryPoint->_pInterpreterData;
+}
+
+void PortableEntryPoint::SetInterpreterData(TADDR addr, PCODE interpreterData)
+{
+    STANDARD_VM_CONTRACT;
+
+    PortableEntryPoint* portableEntryPoint = ToPortableEntryPoint(addr);
+    _ASSERTE(portableEntryPoint->_pInterpreterData == NULL);
+    portableEntryPoint->_pInterpreterData = (void*)PCODEToPINSTR(interpreterData);
+}
+
+PortableEntryPoint* PortableEntryPoint::ToPortableEntryPoint(TADDR addr)
+{
+    LIMITED_METHOD_CONTRACT;
+    _ASSERTE(addr != NULL);
+
+    PortableEntryPoint* portableEntryPoint = (PortableEntryPoint*)addr;
+    _ASSERTE(portableEntryPoint->_canary == CANARY_VALUE);
+    return portableEntryPoint;
+}
+
+void PortableEntryPoint::Init(MethodDesc* pMD)
+{
+    LIMITED_METHOD_CONTRACT;
+    INDEBUG(_canary = CANARY_VALUE);
+    _pMD = pMD;
+    _pInterpreterData = NULL;
+    _pActualCode = NULL;
+}
+
+InterleavedLoaderHeapConfig s_stubPrecodeHeapConfig;
+
+void StubPrecode::Init(StubPrecode* pPrecodeRX, TADDR secretParam, LoaderAllocator *pLoaderAllocator, TADDR type, TADDR target)
+{
+    LIMITED_METHOD_CONTRACT;
+    _ASSERTE(!"StubPrecode::Init is not supported with Portable EntryPoints");
```

Comment:
Should all these be WASM-TODOs? I think we want to ifdef out the callers to do the right thing as a follow up so that we do not have to be dealing them one at a time forever.

Reply:
I agree that these stubs that just assert are the right approach for now. I am just saying that we do not want these stubs to stay. We want to get rid of all of them before we declare this done.

Reply:
> I agree that these stubs that just assert are the right approach for now. I am just saying that we do not want these stubs to stay. We want to get rid of all of them before we declare this done. We have been marking places that require further attention with WASM-TODOs.

+1. My plan is to drive all these to ground. Not necessarily in this PR, but I'm personally not done until these stubbed out functions are all gone.